### PR TITLE
Attempt to register and acquire a shard again during Loading if the first attempt fails with `ErrShardUnknown`

### PR DESCRIPTION
**Pull request by aarshkshah1992**
@dirkmc As discussed yesterday.

### CODE REVIEW

File: retrievalmarket/storage_retrieval_integration_test.go [hunk 1]
```diff
@@ -383,18 +383,6 @@ func newDagStore(t *testing.T, providerNode *testnodes2.TestRetrievalProviderNod
 	return dagStoreWrapper
 }
 
-func newRetrievalHarness(
```
Comment (aarshkshah1992) [inline retrievalmarket/storage_retrieval_integration_test.go:386]: Dead code.

File: dagstore/dagstorewrapper.go [hunk 2]
```diff
@@ -110,8 +112,23 @@ func (ds *dagStoreWrapper) LoadShard(ctx context.Context, pieceCid cid.Cid) (car
 	key := shard.KeyFromCID(pieceCid)
 	resch := make(chan dagstore.ShardResult, 1)
 	err := ds.dagStore.AcquireShard(ctx, key, resch, dagstore.AcquireOpts{})
+
 	if err != nil {
-		return nil, xerrors.Errorf("failed to schedule acquire shard for piece CID %s: %w", pieceCid, err)
+		if errors.Unwrap(err) != dagstore.ErrShardUnknown {
```
Comment (dirkmc) [inline dagstore/dagstorewrapper.go:117]: nit: can we use `xerrors.Is(err, dagstore.ErrShardUnknown)?`

**Review by dirkmc — APPROVED** [review]
LGTM 👍

File: dagstore/dagstorewrapper.go (hunk 2 above)
Comment (aarshkshah1992) [reply dagstore/dagstorewrapper.go:117]: @dirkmc Is there a semantic difference between the two ?

Comment (aarshkshah1992) [reply dagstore/dagstorewrapper.go]: Done.

**Pull request merged**